StorageUtility/Linux/StorageUtility.cpp

```diff
@@ -17,251 +17,181 @@ limitations under the License.
-//    };
+    GetStorageDevicePaths(devicePaths, sysDevicePaths);
+    for (int i = 0; i < devicePaths.size(); ++i)
+    {
+        std::shared_ptr<vtStor::cDevice> device = std::make_shared<cDevice>(devicePaths[i], sysDevicePaths[i]);
+        Devices.push_back(device);
+    }
+}
 
-//    error = EnumerateDevices(callBack, &GUID_DEVINTERFACE_DISK, OnErrorBehavior);
-    return(error);
+bool IsAtaDeviceBus(udev_device* UdevDevice)
+{
+    const char* bus = udev_device_get_property_value(UdevDevice, "ID_BUS");
+    if (nullptr == bus)
+    {
+        return false;
+    }
+    if (0 == (strcmp(bus, "ata")) || (0 == strcmp(bus, "sata")))
+    {
+        return true;
+    }
 }
 
+bool IsScsiDeviceBus(udev_device* UdevDevice)
+{
+    //! TODO Check Disk has Scsi Adapter
+    return false;
+}
 
-/*
-eErrorCode GetStorageDevicePaths( std::vector<String>& Paths, eOnErrorBehavior OnErrorBehavior )
+bool IsAtaDeviceBus(sStorageAdapterProperty StorageAdapterProperty)
 {
-    return( GetDevicePaths( Paths, &GUID_DEVINTERFACE_DISK, OnErrorBehavior ) );
+    return StorageAdapterProperty.AtaBus;
 }
-*/
 
-/*eErrorCode GetDevicePaths( std::vector<String>& Paths, const GUID* InterfaceClassGUID, eOnErrorBehavior OnErrorBehavior )
+bool IsScsiDeviceBus(sStorageAdapterProperty StorageDeviceProperty)
 {
-    eErrorCode error = eErrorCode::None;
+    //! TODO: Check StorageDeviceProperty is SCSI type
+}
 
-    sEnumerateDevicesCallback callBack;
-    callBack.Data = (void*)&Paths;
-    callBack.Function = [](void* Data, const HDEVINFO& DevInfoHandle, SP_DEVINFO_DATA& DevInfoData, SP_DEVICE_INTERFACE_DATA& DevInterfaceData, const PSP_INTERFACE_DEVICE_DETAIL_DATA& DevDetailData, U32 SizeOfDevDetailData, eErrorCode& ErrorCode)
+bool IsStorageDisk(udev_device* UdevDevice)
+{
+    const char* udevDeviceType = udev_device_get_devtype(UdevDevice);
+    if (0 == strcmp(udevDeviceType, "disk"))
     {
-        std::vector<String>* paths = ( std::vector<String>* )Data;
-        paths->push_back( DevDetailData->DevicePath );
-    };
+        if (true == IsAtaDeviceBus(UdevDevice) || true == IsScsiDeviceBus(UdevDevice))
+        {
+            return true;
+        }
+    }
 
-    error = EnumerateDevices( callBack, InterfaceClassGUID, OnErrorBehavior );
-    return( error );
-}*/
+    return false;
+}
 
-/*eErrorCode EnumerateDevices( sEnumerateDevicesCallback& Callback, const GUID* InterfaceClassGUID, eOnErrorBehavior OnErrorBehavior )
+eErrorCode GetStorageDevicePaths(std::vector<String>& DevicePaths, std::vector<String>& SysDevicePaths)
 {
-    HDEVINFO devInfoHandle =
-        SetupDiGetClassDevs
-        (
-        InterfaceClassGUID,
-        NULL,
-        NULL,
-        DIGCF_PRESENT | DIGCF_DEVICEINTERFACE
-        );
-    if ( INVALID_HANDLE_VALUE == devInfoHandle )
+    struct udev *udevObject;
+    struct udev_enumerate *enumerate;
+    struct udev_list_entry *devices, *udevListEntry;
+    struct udev_device *udevDevice;
+
+    //! Create the udev object
+    //!
```

Comment:
Use this below if commenting on the function
//! ,,,,
//!
if you are commenting inside the function, just use //!